File: src/com/douma/_10_day_栈和队列/单调栈/LeftLastSmaller.cpp
public:
    vector<int> findRightSmall(vector<int> &nums) {
        vector<int> ans(nums.size());
        stack<int> st;
        // 从右往左遍历
        for (size_t i = nums.size() - 1; i >= 0; i--) {
            int x = nums[i];
            // 单调递增栈
            while (!st.empty() && x < nums[st.peek()]) {
                ans[st.top()] = i;
                st.pop();
            }
            st.push(i);
        }
        while (!st.empty()) {
            ans[st.top()] = -1;
            st.pop();
        }
        return ans;
    }

File: src/com/douma/_10_day_栈和队列/单调栈/LeftLastSmaller.js
var findRightSmall = function(nums) {
    const ans = new Array(nums.length).fill(-1)
    const stack = []
    // 从右往左遍历
    for (let i = nums.length - 1; i >= 0; i--) {
        const x = nums[i]
        // 单调递增栈
        while (stack.length && x < nums[stack[stack.length - 1]]) {
            ans[stack[stack.length - 1]] = i
            stack.pop()
        }
        stack.push(i)
    }
    return ans
}